libtenzir/builtins/formats/json.cpp

```diff
@@ -1912,6 +1914,64 @@ class write_ndjson_plugin final : public virtual operator_plugin2<write_json> {
   }
 };
 
+class print_json_plugin : public virtual function_plugin {
+public:
+  auto name() const -> std::string override {
+    return compact_ ? "print_ndjson" : "print_json";
+  }
+
+  print_json_plugin(bool compact) : compact_{compact} {
+  }
+
+  auto make_function(invocation inv,
+                     session ctx) const -> failure_or<function_ptr> override {
+    auto expr = ast::expression{};
+    auto args = printer_args{};
+    auto parser = argument_parser2::function(name());
+    parser.positional("x", expr, "any");
+    args.add(parser, false, false, false);
+    TRY(parser.parse(inv, ctx));
+    auto opts = json_printer_options{
+      .tql = false,
+      .style = no_style(),
+      .oneline = compact_,
+      .omit_null_fields = args.omit_null_fields or args.omit_all,
+      .omit_nulls_in_lists = args.omit_nulls_in_lists or args.omit_all,
+      .omit_empty_records = args.omit_empty_objects or args.omit_all,
+      .omit_empty_lists = args.omit_empty_lists or args.omit_all,
+    };
+    return function_use::make(
+      [call = inv.call.get_location(), printer = tenzir::json_printer{opts},
+       expr = std::move(expr)](evaluator eval, session) {
+        return map_series(eval(expr), [&](series values) -> multi_series {
+          if (values.type.kind().is<null_type>()) {
+            return values;
```

Comment:
This shouldn't short-circuit, but rather print a series of strings containing the text `null`.